src/Config.php

```diff
@@ -27,40 +29,49 @@ public function get(string $key, mixed $default = null): mixed
     /**
      * @throws InvalidConfigException
      */
-    private function getByKeyPath(array $key)
+    private function getByKeyPath(array $key): mixed
     {
-        $file = sprintf('%s/%s.php', $this->directory, $key[0]);
+        $subDirectory = array_shift($key);
 
-        if (!file_exists($file)) {
+        $file = realpath("{$this->directory}/{$subDirectory}.php");
```

Comment:
`$file` is not a user-facing string, strictly internal. No reason for it to be formatted nicely.

Apart from that, also quite bad for performance

With realpath: 0.98409108161926
Without realpath: 0.92173115730286

(Seconds, lower is better. Time for 100.000 lookups, average over 25 runs)